src/include/DuckUtils.h

```diff
@@ -49,6 +49,11 @@ String createUuid(int length = CDPCFG_UUID_LEN);
  */
 String convertToHex(byte* data, int size);
 
+template<typename T>
+String toString(const std::vector<T> & vec) {
+  return std::string(vec.begin(), vec.end()).c_str();
```

Comment:
Is the c_str() member function call necessary for the return? I'm assuming that `String` datatype is `std::string`

Reply:
It would be good to remove unneeded code, but it's necessary here. (`String` is the Arduino string, while `std::string` is the C++ standard library, so they're not the same thing. I used std::string because it has a constructor that takes the start and end iterators.)

When I remove the .c_str(), I get:

```
In file included from /Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/include/../CdpPacket.h:5:0,
                 from /Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/include/DuckPacket.h:4,
                 from /Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/DuckPacket.cpp:1:
/Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/include/../include/DuckUtils.h: In instantiation of 'String duckutils::toString(const std::vector<_RealType>&) [with T = unsigned char]':
/Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/DuckPacket.cpp:103:3:   required from here
/Users/rolsen/Documents/Arduino/libraries/ClusterDuck-Protocol/src/include/../include/DuckUtils.h:60:44: error: could not convert 'std::__cxx11::basic_string<char>((& vec)->std::vector<_Tp, _Alloc>::begin<unsigned char, std::allocator<unsigned char> >(), (& vec)->std::vector<_Tp, _Alloc>::end<unsigned char, std::allocator<unsigned char> >(), std::allocator<char>())' from 'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' to 'String'
   return std::string(vec.begin(), vec.end());
                                            ^
```

Reply:
Oops, just saw Charlie's comment below.